pkg/controller/machine/machine.go

```diff
@@ -614,16 +614,23 @@ func (r *Reconciler) deleteNodeForMachine(machine *clusterv1alpha1.Machine) erro
 	if machine.Status.NodeRef != nil {
 		objKey := ctrlruntimeclient.ObjectKey{Name: machine.Status.NodeRef.Name}
 		node := &corev1.Node{}
+		nodeFound := true
 		if err := r.client.Get(r.ctx, objKey, node); err != nil {
```

Comment:
I think it's more natural to express this in the opposite way:

```
if err := r.client.Get(r.ctx, objKey, node); err == nil {
			if err := r.client.Delete(r.ctx, node); err != nil {
				if !kerrors.IsNotFound(err) {
					return err
				}
				klog.V(2).Infof("node %q does not longer exist for machine %q", machine.Status.NodeRef.Name, machine.Spec.Name)
			}
} else {
			if !kerrors.IsNotFound(err) {
				return err					return fmt.Errorf("failed to get node %s: %v", machine.Status.NodeRef.Name, err)
			}				}
}
```

You don't need the boolean

Reply:
Don't look at the indentation...